### PR TITLE
feat: refactor reviewers

### PR DESCRIPTION
**Pull request by trowik**
use assignees with reviewer role instead of project reviewers

### CODE REVIEW

**Comment by trowik** [comment]
The 3 requests we do for the assignees to filter the reviewers in analysis page will be changed later on. However we don't have the needed filters in backend at the moment, so we had to implement a workaround. The filters will be added later on and that part will be refactored.

File: app/abilities/report.js [hunk 1]
```diff
@@ -4,19 +4,26 @@ import { Ability } from "ember-can";
...
+      const isReviewer =
+        (this.get("model.taskAssignees") ?? [])
+          .mapBy("user.id")
+          .includes(this.get("user.id")) ||
+        (this.get("model.projectAssignees") ?? [])
+          .mapBy("user.id")
+          .includes(this.get("user.id")) ||
+        (this.get("model.customerAssignees") ?? [])
+          .mapBy("user.id")
+          .includes(this.get("user.id"));
```
Comment (aziiee) [inline app/abilities/report.js:25]: Get rid of redundant code maybe like this:

```js
const reviewer = (this.get("model.taskAssignees") ?? [])
        .concat(
          this.get("model.projectAssignees") ?? [],
          this.get("model.customerAssignees") ?? []
        )
        .mapBy("user.id")
        .includes(this.get("user.id"));
```

File: app/analysis/index/controller.js [hunk 2]
```diff
@@ -291,8 +291,42 @@ const AnalysisController = Controller.extend(AnalysisQueryParams.Mixin, {
...
+
+    const projectAssignees = (yield this.store.query("project-assignee", {
+      is_reviewer: 1
+    })).toArray();
+    const taskAssignees = (yield this.store.query("task-assignee", {
+      is_reviewer: 1
+    })).toArray();
+    const customerAssignees = (yield this.store.query("customer-assignee", {
+      is_reviewer: 1
+    })).toArray();
```
Comment (aziiee) [inline app/analysis/index/controller.js:305]: Is `toArray()` really needed?

File: app/models/customer-assignee.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * @class CustomerAssignee
+ * @extends DS.Model
+ * @public
+ */
+export default Model.extend({
+  /**
+   * The user
+   *
+   * @property user
```
Comment (aziiee) [inline app/models/customer-assignee.js:21]: customer

File: app/models/project-assignee.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * @class ProjectAssignee
+ * @extends DS.Model
+ * @public
+ */
+export default Model.extend({
+  /**
+   * The user
+   *
+   * @property user
```
Comment (aziiee) [inline app/models/project-assignee.js:21]: project

File: app/models/task-assignee.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * @class TaskAssignee
+ * @extends DS.Model
+ * @public
+ */
+export default Model.extend({
+  /**
+   * The user
+   *
+   * @property user
```
Comment (aziiee) [inline app/models/task-assignee.js:21]: task

File: tests/acceptance/project-test.js [hunk 6]
```diff
@@ -12,10 +12,11 @@ module("Acceptance | projects", function(hooks) {
 
   hooks.beforeEach(async function() {
     const user = this.server.create("user", { isReviewer: true });
+    this.server.create("project-assignee", {
+      user
+    });
     // eslint-disable-next-line camelcase
     await authenticateSession({ user_id: user.id });
-
-    this.server.create("project", { reviewers: [user] });
```
Comment (aziiee) [inline tests/acceptance/project-test.js:18]: Insert this again. This is the reason for the failing acceptance test of projects. If we do not have any projects, there won't be any to select from.

**Review by aziiee — CHANGES_REQUESTED** [review]

File: tests/acceptance/project-test.js (hunk 6 above)
Comment (trowik) [reply tests/acceptance/project-test.js:18]: A project is created via the project-assignee factory, am I right?

File: app/analysis/index/controller.js (hunk 2 above)
Comment (trowik) [reply app/analysis/index/controller.js]: Nope

File: app/abilities/report.js (hunk 1 above)
Comment (trowik) [reply app/abilities/report.js]: Works

File: tests/acceptance/project-test.js (hunk 6 above)
Comment (aziiee) [reply tests/acceptance/project-test.js:18]: Exactly

**Review by aziiee — APPROVED** [review]

**Pull request merged**